pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -635,7 +637,13 @@ func (r *Reconciler) fetchRunningInstancesInInstanceGroup(projectID string, zone
 		},
 	)
 	if err != nil {
-		return nil, fmt.Errorf("instanceGroupsListInstances request failed: %v", err)
+		if errRegister := r.registerNewInstanceGroup(); errRegister != nil {
+			return nil, fmt.Errorf("instanceGroupsListInstances request failed: %v, failed to register the new instance group named %v: %v", err, instaceGroup, errRegister)
+		}
+		if errBackendService := r.updateBackendServiceWithInstanceGroup(); errBackendService != nil {
```

Comment:
Use %s for the instance group substitution if it's a string, else try and get the name of the instance group from the object